data-prepper-plugins/peer-forwarder/src/test/java/com/amazon/dataprepper/plugins/processor/peerforwarder/PeerForwarderTest.java

```diff
@@ -57,6 +60,23 @@
     private static final ExportTraceServiceRequest REQUEST_3 = generateRequest(SPAN_1, SPAN_2, SPAN_3);
     private static final ExportTraceServiceRequest REQUEST_4 = generateRequest(SPAN_4, SPAN_5, SPAN_6);
 
+    private MockedStatic<PeerClientPool> peerClientPoolClassMock;
+
+    @Mock
+    private PeerClientPool peerClientPool;
+
+    @Before
+    public void setUp() {
+        peerClientPoolClassMock = Mockito.mockStatic(PeerClientPool.class);
+        peerClientPoolClassMock.when(PeerClientPool::getInstance).thenReturn(peerClientPool);
+    }
```

Comment:
One way to get around having to mock static here would be using the multi-parameter constructor to build the test object, something like:
```
    @Before
    public void setup() {
        peerForwarder =  new PeerForwarder(mockPluginSetting,
                mockPeerForwarder,
                mockHashing,
                maxRequestsValue);
    }
```
This keeps the unit test from having to know how to construct a PeerForwarder (that functionality is currently being handled/tested in PeerForwarderConfig and PeerForwarderConfigTest).

Doesn't have to change in this PR, just a thought (also my opinion, I'm sure some would prefer to test actual objects when possible but there are pros/cons to each).

Reply:
> This keeps the unit test from having to know how to construct a PeerForwarder (that functionality is currently being handled/tested in PeerForwarderConfig and PeerForwarderConfigTest).

Agreed on that the design will make testing config and main class more independent, but I would assume here PeerForwarderConfig should work fine. Otherwise it should be captured in PeerForwarderConfigTest.

Also, I am not sure how it relates to avoiding PeerClientPool static mock since it is still a singleton we want to reset for each test case.